lib/Backends/CMakeLists.txt

```diff
@@ -17,31 +17,28 @@ endif()
 
 add_library(Backend
               Backend.cpp
+              BackendUtils.cpp
               TraceEvents.cpp)
 target_link_libraries(Backend
                       PRIVATE
                         Base
+                        CodeGen
                         Graph
                         IR
                         Optimizer)
 
 add_library(Backends
               Backends.cpp)
 target_link_libraries(Backends
+                      PUBLIC
+                        Backend
                       PRIVATE
-                        BackendUtils
                         Base
                         Graph
                         Interpreter
                         Optimizer
                         ${linked_backends})
 
-add_library(BackendUtils BackendUtils.cpp)
```

Comment:
@nickgg what was the cause for this change? I'd really like to detangle all of the dependencies here in `Backends/` but for example `~CompiledFunction()` which is defined `CompiledFunction.h` calls `freeConstants` defined in `BackendUtils.cpp`. This is breaking shared libraries if I try to remove the unnecessary dependency on `Backend` by `Backends` but I can't fix it because `BackendUtils` library is gone now.

Reply:
Actually this is probably better and we can just depend on `Backends` and `Backend` wherever both are needed elsewhere (example onnxifi, unit tests, etc) and that way we can remove the dependency of `Backend` by `Backends` and make the dependencies cleaner here.